### PR TITLE
go-sdk: Pipeline notification changes

### PR DESCRIPTION
**Pull request by blinktag**
* Bump protos to v0.1.21
* Update `NotifyRequest` with new fields
* Fix README.md example

### CODE REVIEW

File: sdks/go/go_sdk.go [hunk 1]
```diff
@@ -874,6 +878,7 @@ type condition struct {
 
 // handleCondition is a wrapper for inspecting the step condition and potentially
 // performing a notification and injecting metadata back into the response.
+// TODO: simplify these params
```
Comment (dselans) [inline sdks/go/go_sdk.go:881]: Agreed - this is getting really long. I have to work on the SDK next, I'll do this.

File: sdks/go/server/client.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ type IServerClient interface {
 	NewAudience(ctx context.Context, aud *protos.Audience, sessionID string) error
 
 	// Notify calls to streamdal server to trigger the configured notification rules for the specified step
-	Notify(ctx context.Context, pipeline *protos.Pipeline, step *protos.PipelineStep, aud *protos.Audience) error
+	Notify(ctx context.Context, pipeline *protos.Pipeline, step *protos.PipelineStep, aud *protos.Audience, payload []byte, conditionType protos.NotifyRequest_ConditionType) error
```
Comment (dselans) [inline sdks/go/server/client.go:37]: Really good that we can specify concrete types.

**Review by dselans — APPROVED** [review]
Let's get this merged!

**Review by dselans — CHANGES_REQUESTED** [review]
Undo the README.md changes - I had to update it a bit ago because of the SDK migration.

**Comment by dselans** [comment]
I rebased on main + removed the README updates.

**Review by dselans — APPROVED** [review]

File: sdks/go/README.md [hunk 3]
```diff
@@ -40,18 +40,23 @@ import (
 )
 
 func main() {
+	ctx, cancel := context.WithCancel(context.Background())
```
[the remark this answers is not in the document]
Comment (blinktag) [reply sdks/go/README.md]: Sweet! Done

**Comment by blinktag** [comment]
Waiting for frontend changes to land before merging

**Pull request merged**